### PR TITLE
Update for metrodr stretch cluster documentation

### PR DESCRIPTION
**Pull request by netzzer**

### CODE REVIEW

File: training/modules/ocs4/pages/ocs4-metro-stretched.adoc [hunk 1]
```diff
@@ -0,0 +1,739 @@
...
+----
+oc get deployment -o yaml -n my-shared-storage | less
+----
+
+Search for `containers` and repeat the search a few times until your output is similar. There is no pod placement rules in the *Deployment*.
+
+.Example Output:
+[source,yaml]
+----
+[...]
```
Comment (jeanchlopez) [inline training/modules/ocs4/pages/ocs4-metro-stretched.adoc:594]: Given there are 2 spec entries in the deployment would be better to highlight like this
  template:
    metadata:
      annotations:
        openshift.io/generated-by: OpenShiftNewApp
      creationTimestamp: null
      labels:
        deployment: file-uploader
    spec:             <--- Start insert after this line of the following boilded lines
        topologySpreadConstraints:
          - labelSelector:
              matchLabels:
                deployment: file-uploader
            maxSkew: 1
            topologyKey: topology.kubernetes.io/zone
            whenUnsatisfiable: DoNotSchedule
          - labelSelector:
               matchLabels:
                 deployment: file-uploader
            maxSkew: 1
            topologyKey: kubernetes.io/hostname
            whenUnsatisfiable: ScheduleAnyway
        nodeSelector:
          node-role.kubernetes.io/worker: ""
      containers: <--- Next line afte rinsert
      - image: image-registry.openshift-image-registry.svc:5000/my-shared-storage/file-uploader@sha256:b198c4e9e4a22a96cea7599c3b3c7d600b813ccdd8d79f373ae6c0e4f5979389
        imagePullPolicy: IfNotPresent
        name: file-uploader
        ports:

File: training/modules/ocs4/pages/ocs4-metro-stretched.adoc [hunk 2]
```diff
@@ -0,0 +1,739 @@
...
+                 deployment: file-uploader
+            maxSkew: 1
+            topologyKey: kubernetes.io/hostname
+            whenUnsatisfiable: ScheduleAnyway
+        nodeSelector:
+          node-role.kubernetes.io/worker: ""
+        containers:
+        - image: image-registry.openshift-image-registry.svc:5000/my-shared-storage/file-uploader@sha256:a458ea62f990e431ad7d5f84c89e2fa27bdebdd5e29c5418c70c56eb81f0a26b
+          imagePullPolicy: IfNotPresent
+          name: file-uploader
```
Comment (jeanchlopez) [inline training/modules/ocs4/pages/ocs4-metro-stretched.adoc:631]: Also prepared a file that does this with one apply but we need to test ot works on your end at https://gist.github.com/jeanchlopez/73335e6c6e02337f988a18f56d5afb03

File: training/modules/ocs4/pages/ocs4-metro-stretched.adoc [hunk 3]
```diff
@@ -0,0 +1,739 @@
...
+The following will be installed:
+
+- The ODF Operator (OCS Operator in OCP Web console)
+- All other ODF resources (Ceph Pods, NooBaa Pods, StorageClasses)
+
+=== ODF Operator Deployment
+
+Start with creating the `openshift-storage` namespace.
+
+[source,role="execute"]
```
[the remark this answers is not in the document]
Comment (netzzer) [reply training/modules/ocs4/pages/ocs4-metro-stretched.adoc:169]: I used your graphic that shows the openshift-storage namespace already created - https://github.com/red-hat-storage/ocs-training/blob/master/training/modules/ocs4/images/OCS4-4.7-OCP4-OperatorHub-Subscribe.png.

File: training/modules/ocs4/pages/ocs4-metro-stretched.adoc [hunk 4]
```diff
@@ -0,0 +1,739 @@
...
+:source-language: shell
+:numbered:
+// Activate experimental attribute for Keyboard Shortcut keys
+:experimental:
+:source-highlighter: pygments
+:hide-uri-scheme:
+
+== Overview
+
+In order to provide a level of disaster recovery an OpenShift Container Platform (OCP) deployment can be `stretched` between two geographically different locations. To do this necessary services including storage must be able to survive when one of the two locations is partially or totally not available. This solution sometimes is called `Metro DR stretch cluster` which means the distance between the sites needs to be limited so that OpenShift services and storage can operate with acceptable performance.
```
[the remark this answers is not in the document]
Comment (netzzer) [reply training/modules/ocs4/pages/ocs4-metro-stretched.adoc:14]: Changed ...

To be resilient in the face of a disaster necessary OCP services, including storage, must be able to survive when one of the two locations is partially or totally not available.

File: training/modules/ocs4/pages/ocs4-metro-stretched.adoc [hunk 5]
```diff
@@ -0,0 +1,739 @@
...
+----
+.Example output:
+----
+NAME
+perf1-mz8bt-master-2
+perf1-mz8bt-worker-ntkp8
+perf1-mz8bt-worker-qpwsr
+----
+
+You have now created the `Metro DR stretch cluster` topology zones that are required for site outage recovery.
```
[the remark this answers is not in the document]
Comment (netzzer) [reply training/modules/ocs4/pages/ocs4-metro-stretched.adoc:99]: fixed ...

The `Metro DR stretch cluster` topology zone labels are now applied to the appropriate OCP nodes to define the three locations. Next step is installing the storage operators from OCP *OperatorHub*.

File: training/modules/ocs4/pages/ocs4-metro-stretched.adoc (hunk 1 above)
Comment (netzzer) [reply training/modules/ocs4/pages/ocs4-metro-stretched.adoc:594]: Attempted fix

File: training/modules/ocs4/pages/ocs4-metro-stretched.adoc [hunk 6]
```diff
@@ -0,0 +1,739 @@
...
+To validate the labels run the following commands using the example labels for the 3 zones:
+
+[source,role="execute"]
+----
+oc get nodes -l topology.kubernetes.io/zone=arbiter | awk {'print $1}'
+----
+.Example output:
+----
+NAME
+perf1-mz8bt-master-0
```
[the remark this answers is not in the document]
Comment (netzzer) [reply training/modules/ocs4/pages/ocs4-metro-stretched.adoc:72]: retested and `node/` is not present from query:

`oc get nodes -l topology.kubernetes.io/zone=arbiter | awk {'print $1}'`

File: training/modules/ocs4/pages/ocs4-metro-stretched.adoc (hunk 2 above)
Comment (netzzer) [reply training/modules/ocs4/pages/ocs4-metro-stretched.adoc:631]: Going to leave as edit for now. Used your content to update `file-uploader-zone-deployment.yaml` so can be tested/used later.

File: training/modules/ocs4/pages/ocs4-metro-stretched.adoc [hunk 7]
```diff
@@ -0,0 +1,739 @@
...
+NOTE: Currently `Metro DR stretch cluster` is designed to be deployed where latencies do not exceed 4 milliseconds round-trip time (RTT) between locations. Contact https://access.redhat.com/support[Red Hat Customer Support] if you are planning to deploy with higher latencies.
+
+The intent of this solution guide is to detail the steps and commands necessary to deploy `OpenShift Data Foundation`(ODF) along with Kubernetes zone topology labels to achieve a highly available platform.
+
+This is a general overview of the steps required to configure and execute `Metro DR stretch cluster` capabilities using ODF *4.7* or greater.
+
+[start=1]
+. *Ensure you have at least 3 OCP master nodes in 3 different locations* +
+Ensure that there is a master node in each of the two locations and that the 3rd master node is in a 3rd location to act as the arbiter in the case of site outage.
+. *Ensure you have at least 4 OCP worker nodes in 2 different locations* +
```
[the remark this answers is not in the document]
Comment (netzzer) [reply training/modules/ocs4/pages/ocs4-metro-stretched.adoc:25]: . *Ensure you have a minimum of 4 OCP worker nodes* +
Ensure you have at least 4 OCP worker nodes evenly dispersed across 2 different locations.

File: training/modules/ocs4/pages/ocs4-metro-stretched.adoc [hunk 8]
```diff
@@ -0,0 +1,739 @@
...
+Ensure that the worker nodes are evenly split in each of the two locations with a minimum of 2 worker nodes in each location.
+. *Assign kubernetes topology zone labels to each master and worker node* +
+Assign unique topology.kubernetes.io/zone labels to each node to define failure domains as well as the arbiter zone.
+. *Install OpenShift Container Storage operator* +
+Install the OCS operator using OperatorHub in the OpenShift Web console.
```
[the remark this answers is not in the document]
Comment (netzzer) [reply training/modules/ocs4/pages/ocs4-metro-stretched.adoc:30]: fixed. Issue is it is OCS in OperatorHub, not ODF

. *Install ODF operators and storage cluster* +
Install the OpenShift Container Storage operator and storage cluster using OperatorHub in the OpenShift Web console.

File: training/modules/ocs4/pages/ocs4-metro-stretched.adoc [hunk 9]
```diff
@@ -0,0 +1,739 @@
...
+. *Install a sample application for failure testing* +
+Install an OCP application that can be configured to be highly available with critical components deployed in both data zones.
+. *Power off all OCP nodes in one location* +
+Simulate a site outage by powering off all nodes (including master node) and validate sample application availability.
+. *Power on all OCP nodes in failed location* +
+Simulate a recovery by powering back on all nodes and validate sample application availability.
+
+== Metro DR Stretch Cluster Prerequisites 
+
+The diagram shows the most simple deployment for `Metro DR stretch cluster`. The ODF Monitor pod that is the `arbiter` in case of a site outage can be scheduled on a master node. The diagram does not show an additional master nodes in each `Data Zone` which is required for a highly available OCP control plane. Also, it is critical that the OCP nodes in one location have network reachability to the OCP nodes in the other two locations.
```
[the remark this answers is not in the document]
Comment (netzzer) [reply training/modules/ocs4/pages/ocs4-metro-stretched.adoc:40]: Fixed

File: training/modules/ocs4/pages/ocs4-metro-stretched.adoc [hunk 10]
```diff
@@ -0,0 +1,739 @@
...
+
+=== Apply topology zone labels to OCP nodes
+
+Prior to installing ODF the nodes that will be used for storage and the node for the arbiter function must be labeled to define their zone. In our case we will use the label `datacenter1` for one location with storage nodes and `datacenter2` for the other location. The zone that will have the arbiter function in the case of a site outage will use the `arbiter` label. These labels are arbitrary but they do need to be unique for the 3 locations.
+
+For example, you can label the nodes and in the diagram as follows (including master nodes not shown):
+
+topology.kubernetes.io/zone=arbiter for Master3
+topology.kubernetes.io/zone=datacenter1 for Master1 Worker1 & Worker2
+topology.kubernetes.io/zone=datacenter2 for Master2 Worker3 & Worker4
```
[the remark this answers is not in the document]
Comment (netzzer) [reply training/modules/ocs4/pages/ocs4-metro-stretched.adoc:55]: Fixed

File: training/modules/ocs4/pages/ocs4-metro-stretched.adoc [hunk 11]
```diff
@@ -0,0 +1,739 @@
...
+To apply the labels to the node using `oc` CLI do the following: 
+----
+oc label node <node_name> topology.kubernetes.io/zone=<label>
+----
+
+To validate the labels run the following commands using the example labels for the 3 zones:
+
+[source,role="execute"]
+----
+oc get nodes -l topology.kubernetes.io/zone=arbiter | awk {'print $1}'
```
[the remark this answers is not in the document]
Comment (netzzer) [reply training/modules/ocs4/pages/ocs4-metro-stretched.adoc:67]: took suggestion, fixed.

File: training/modules/ocs4/pages/ocs4-metro-stretched.adoc [hunk 12]
```diff
@@ -0,0 +1,739 @@
...
+// Activate experimental attribute for Keyboard Shortcut keys
+:experimental:
+:source-highlighter: pygments
+:hide-uri-scheme:
+
+== Overview
+
+In order to provide a level of disaster recovery an OpenShift Container Platform (OCP) deployment can be `stretched` between two geographically different locations. To do this necessary services including storage must be able to survive when one of the two locations is partially or totally not available. This solution sometimes is called `Metro DR stretch cluster` which means the distance between the sites needs to be limited so that OpenShift services and storage can operate with acceptable performance.
+
+NOTE: Currently `Metro DR stretch cluster` is designed to be deployed where latencies do not exceed 4 milliseconds round-trip time (RTT) between locations. Contact https://access.redhat.com/support[Red Hat Customer Support] if you are planning to deploy with higher latencies.
```
[the remark this answers is not in the document]
Comment (netzzer) [reply training/modules/ocs4/pages/ocs4-metro-stretched.adoc:16]: Fixed

**Pull request merged**